test/distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py

```diff
@@ -93,23 +95,40 @@ class TestClipGradNormWorldSize2(_TestClipGradNormBase):
     def world_size(self) -> int:
         return min(torch.cuda.device_count(), 2)
 
-    @skip_if_lt_x_gpu(2)
-    def test_clip_grad_norm_1d(self):
+    def _test_clip_grad_norm_1d(self, offload_policy: OffloadPolicy):
         for norm_type in (2, 1, float("inf")):
             torch.manual_seed(42)
             model_args = ModelArgs(dropout_p=0.0)
             model = Transformer(model_args)
             ref_model = replicate(copy.deepcopy(model).cuda())
             ref_optim = torch.optim.Adam(ref_model.parameters(), lr=1e-2)
+
             for module in model.modules():
                 if isinstance(module, TransformerBlock):
-                    fully_shard(module)
-            fully_shard(model)
+                    fully_shard(module, offload_policy=offload_policy)
+            fully_shard(model, offload_policy=offload_policy)
             optim = torch.optim.Adam(model.parameters(), lr=1e-2)
             inp = torch.randint(0, model.model_args.vocab_size, (3, 16), device="cuda")
-            self._test_clip_grad_norm(
-                1, norm_type, ref_model, ref_optim, model, optim, inp
-            )
+
+            if offload_policy == CPUOffloadPolicy() and get_backend() == "nccl":
+                with self.assertRaisesRegex(
+                    RuntimeError,
+                    "FSDP CPU offloading requires backend supporting cpu tensors.",
```

Comment:
nit: consider checking for only keywords rather than full message, so that other developers have more flexibility in changing the error message.